added fucntionality to toggle readonly for env store

louis-murray opened the pull request:
Hello, 

I saw that there was a readOnly option in the env store but no way to set it when calling `nconf.env`. This PR adds that option. I've come across use cases where variables we're being passed in from the environment but then for one reason or another needed to be altered later while the app is still running.

mhamann commented:
@louis-murray thanks for the contribution. A few questions.

Have you looked at the `transform` capability in the current `master` branch? Should be available on npm within a week or so. Would that meet your needs?

One disadvantage here is that you could inadvertently modify a value without meaning to and then introduce subtle bugs into the application, even though you think certain values are being passed in. What specific use cases does making the `env` store writable solve?

louis-murray commented:
@mhamann Hi! Thanks for your feedback. I don't think that transform is applicable to this case. I initially came across this when testing code where I was changing the value of a field set initially through an env var using `nconf.set`. (although I wouldn't prompt this change just because of that) Besides that though I could see cases where a config could be initially set via env variables and then need to be change later, maybe like a key/token that has a limited lifespan or something like that. Mainly, I thought this would be a good change to the base because it just helps to cover those potential extra use cases and make nconf more adaptable. Although I do see where you're coming from with accidentally changing a value. With the way I wrote this the default is to still have nconf.env be `readOnly` only changing when the user defines it as such as an option.  I feel like it would be better to provide the option and note in docs the potential case you outlined than just restrict it.

mhamann commented:
After thinking more about this, I do think I see the value in enabling this. I think it also would be useful in the `argv` store if we're going to do it for `env`.

mhamann commented on test/stores/env-test.js:
```diff
@@ -19,5 +19,15 @@ vows.describe('nconf/stores/env').addBatch({
+  },
+  "An instance of nconf.Env with readOnly option set to false": {
+    topic: new nconf.Env({readOnly: false}),
+    "should have it's readOnly property set to false": function (env) {
```
@louis-murray in this test, you should also assert that calling `.set()` actually changes the value.

Additionally, it would be good to modify an existing `readOnly` test to ensure that calling `.set()` fails to modify a value.

louis-murray commented:
@mhamann I added one test to complete-test.js to test that you could use `.set` when `readOnly:false` and updated one of the other tests using env when `readOnly:true` to check that works as expected. I can update the argv store with toggle-able readOnly as part of this PR too if you'd like.

mhamann commented:
@louis-murray LGTM. Let's go ahead and add to argv as well (along with requisite tests, of course ;-)) and then we'll call this good to go.

Thanks very much!

louis-murray commented:
@mhamann done! Thanks for the feedback on my code and everything!

mhamann commented:
Thank you! This should land as part of v1.0.
